### PR TITLE
Add ignoreInWatchMode to discovered

### PR DESCRIPTION
**Pull request by adamiak**
Resolves L2B-5659

After introduction of template suggestions (`field.target.template`), the main config.jsonc will not always define explicitly which templates will be used, so it's no longer possible to directly read `ignoreInWatchMode` from it for every discovered contract. In the future, this value might be potentially overwritten/extended, meaning it must be gathered from multiple places / template files. That's why `ignoreInWatchMode` is added to discovered.json in its final form.

This PR also follows the direction of having all relevant data in discovered.json, removing requirement of reading config and template files for any subsequent processing of discovery results.

### CODE REVIEW

File: packages/backend/src/modules/implementation-change-report/api/ImplementationChangeController.ts
Comment (mateuszradomski) [inline packages/backend/src/modules/implementation-change-report/api/ImplementationChangeController.ts:1]: Higher up from this place we read all the configs which is no longer necessary, we use the config only for `config.name` which could be replaced with `project`. This allows us to not read the config at all in this file.

File: packages/backend/src/modules/update-monitor/UpdateMonitor.ts
Comment (mateuszradomski) [inline packages/backend/src/modules/update-monitor/UpdateMonitor.ts:1]: What about counting severities? Maybe that should also go to `discovered.json`? Plus, here we can stop reading the configs and just read the `discovered.json` with `readAllProjectsForChain()`.

File: packages/backend/src/modules/update-monitor/api/props/utils/getDiff.ts
Comment (mateuszradomski) [inline packages/backend/src/modules/update-monitor/api/props/utils/getDiff.ts:1]: This function does not need to take the config.

**Review by mateuszradomski — CHANGES_REQUESTED** [review]

File: packages/backend/src/modules/implementation-change-report/api/ImplementationChangeController.ts
Comment (adamiak) [reply packages/backend/src/modules/implementation-change-report/api/ImplementationChangeController.ts:1]: Done

File: packages/backend/src/modules/update-monitor/UpdateMonitor.ts
Comment (adamiak) [reply packages/backend/src/modules/update-monitor/UpdateMonitor.ts:1]: They should go to `discovered.json`, as you suggest. It's going to be done as part of L2B-5690.

**Review by mateuszradomski — APPROVED** [review]

**Pull request merged**